packages/headless/src/features/instant-results/instant-results-slice.ts

```diff
@@ -49,12 +49,16 @@ export const instantResultsReducer = createReducer(
       }
     });
     builder.addCase(fetchInstantResults.fulfilled, (state, action) => {
-      const {results} = action.payload;
       const {cacheTimeout} = action.meta.arg;
       const cached = getCached(state, action.meta);
+
       cached!.isLoading = false;
-      cached!.error = null;
-      cached!.results = results;
+      cached!.error =
```

Comment:
Is this how we differentiate between an expected error (ie: requests happening too fast, which is "normal" with this controller) VS an actual un-expected error that we want to re-surface to the implementer ?

Why only rely on the `message` property of the error ?

Reply:
from my investigations, whenever I `rejectWithValue` in the thunk, it throws an error and I can't catch it.

however, by catching and returning an error
```
    } catch (e) {
      return e as SerializedError;
    }
```
we dispatch a `fulfilled` action with an error as a payload. 
so my thought here as to check whether it's an error by checking if the payload contains a `message`.

Reply:
By doing this, is the below `builder.addCase(fetchInstantResults.rejected, (state, action) => {` still ever called?

Would it be possible to handle the error there instead?